skimage/__init__.py

```diff
@@ -1,79 +1,41 @@
 
 The main package of ``skimage`` only provides a few utilities for converting
-between image data types; for most features, you need to import one of the
-following subpackages:
-
-Subpackages
------------
-color
-    Color space conversion.
-data
-    Test images and example data.
-draw
-    Drawing primitives (lines, text, etc.) that operate on NumPy arrays.
-exposure
-    Image intensity adjustment, e.g., histogram equalization, etc.
-feature
-    Feature detection and extraction, e.g., texture analysis corners, etc.
-filters
-    Sharpening, edge finding, rank filters, thresholding, etc.
-graph
-    Graph-theoretic operations, e.g., shortest paths.
-io
-    Reading, saving, and displaying images and video.
-measure
-    Measurement of image properties, e.g., region properties and contours.
-metrics
-    Metrics corresponding to images, e.g. distance metrics, similarity, etc.
-morphology
-    Morphological operations, e.g., opening or skeletonization.
-restoration
-    Restoration algorithms, e.g., deconvolution algorithms, denoising, etc.
-segmentation
-    Partitioning an image into multiple regions.
-transform
-    Geometric and other transforms, e.g., rotation or the Radon transform.
-util
-    Generic utilities.
-
-Utility Functions
------------------
-img_as_float
-    Convert an image to floating point format, with values in [0, 1].
-    Is similar to `img_as_float64`, but will not convert lower-precision
-    floating point arrays to `float64`.
-img_as_float32
-    Convert an image to single-precision (32-bit) floating point format,
-    with values in [0, 1].
-img_as_float64
-    Convert an image to double-precision (64-bit) floating point format,
-    with values in [0, 1].
-img_as_uint
-    Convert an image to unsigned integer format, with values in [0, 65535].
-img_as_int
-    Convert an image to signed integer format, with values in [-32768, 32767].
-img_as_ubyte
-    Convert an image to unsigned byte format, with values in [0, 255].
-img_as_bool
-    Convert an image to boolean format, with values either True or False.
-dtype_limits
-    Return intensity limits, i.e. (min, max) tuple, of the image's dtype.
-
+between image data types; for most features, you need to import one of its
+subpackages.
 """
 
 __version__ = '0.25.0rc2.dev0'
 
 import lazy_loader as _lazy
 
-__getattr__, __lazy_dir__, _ = _lazy.attach_stub(__name__, __file__)
-
-
-def __dir__():
-    return __lazy_dir__() + ['__version__']
+__getattr__, *_ = _lazy.attach_stub(__name__, __file__)
+
+# `attach_stub` currently ignores __all__ inside the stub file and simply
+# returns every lazy-imported object, so we need to define `__all__` again.
+__all__ = [
+    'color',
+    'data',
+    'draw',
+    'exposure',
+    'feature',
+    'filters',
+    'future',
+    'graph',
+    'io',
+    'measure',
+    'metrics',
+    'morphology',
+    'registration',
+    'restoration',
+    'segmentation',
+    'transform',
+    'util',
+    '__version__',
```

Comment:
Is `__doc__` going away? I recently read something about `__doc__` changing soon, but maybe it was just its parsing...

Reply:
I don't see where `__doc__` was previously included? Am I misunderstanding?

Not sure, but I don't think we need to advertise `__doc__` in `__all__`. It's an implementation detail of Python and it's available at runtime for all intents and purposes, e.g., `help(skimage)`.

I just put `__version__` in here because it might be useful to advertise.

Reply:
Sorry, I wasn't referring to anything 'previously included' or not and, in the context of this PR, I should have!

I was thinking 'in the absolute,' i.e., when I
```py
import skimage as ski
```
and I run
```
ski.[tab]
```
which objects do I expect to be available?